ModularTegustation/tegu_procs.dm

```diff
@@ -46,15 +46,70 @@
 /client/proc/playcreditsmusic(vol = 85)
 	set waitfor = FALSE
 	if(prefs && (prefs.toggles & SOUND_LOBBY))
-		SEND_SOUND(src, sound(returncreditsmusic(), repeat = 0, wait = 0, volume = vol, channel = CHANNEL_LOBBYMUSIC)) // TEGUSTATION: Play end music! SO SAD!
+		SEND_SOUND(src, sound(returncreditsmusic(), repeat = 0, wait = 0, volume = vol, channel = CHANNEL_LOBBYMUSIC))
 
-// Called from sound.dm to find a track to play
-/client/proc/returncreditsmusic() // TEGUSTATION
-	if (prob(15))
-		return 'ModularTegustation/tegusounds/Uhoh_Stinky.ogg'
-	if (prob(5))
-		return 'ModularTegustation/tegusounds/Tegu_Piano_Old.ogg'
-	return 'ModularTegustation/tegusounds/Tegu_Piano.ogg'
+GLOBAL_VAR(roundend_music)
+/client/proc/returncreditsmusic()
+	if(isfile(GLOB.roundend_music)) // If there's a file corresponding to global var path
+		return GLOB.roundend_music
+
+	// If nothing overwrote the music we go deep into the calculations of how well you did
+	var/core_suppression_state = SSlobotomy_corp.core_suppression_state // 0 = Never did; 1 = Initiated; 2 = Completed.
+	var/players = GLOB.joined_player_list.len
+	var/survivors = 0
+	for(var/mob/M in GLOB.mob_list)
+		if(isnewplayer(M))
+			continue
+		if(!M.mind)
+			continue
+		if(ishuman(M) && M.ckey && M.stat != DEAD)
```

Comment:
I should add it to the normal roundend counter, too, since that one will count sentient abnos and other stuff, while we only want to count humans.